packages/mdc-textfield/foundation.js

```diff
@@ -57,17 +59,22 @@ class MDCTextFieldFoundation extends MDCFoundation {
       registerBottomLineEventHandler: () => {},
       deregisterBottomLineEventHandler: () => {},
       getNativeInput: () => {},
-      getBottomLineFoundation: () => {},
-      getHelperTextFoundation: () => {},
     });
   }
 
   /**
    * @param {!MDCTextFieldAdapter=} adapter
+   * @param {!FoundationMapType=} foundationMap Map of all subcomponents to subfoundations.
    */
-  constructor(adapter = /** @type {!MDCTextFieldAdapter} */ ({})) {
+  constructor(adapter = /** @type {!MDCTextFieldAdapter} */ ({}),
+    foundationMap = /** @type {!FoundationMapType} */ ({})) {
     super(Object.assign(MDCTextFieldFoundation.defaultAdapter, adapter));
 
+    /** @type {!MDCTextFieldBottomLineFoundation|undefined} */
```

Comment:
FYI, this is the correct way to annotate an [optional property/param](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#optional-parameter-in-a-param-annotation). I talked to Brendan (our resident Closure Compiler expert) and crawled through the [JS Style Guide](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability) and [Closure Compiler JSDoc wiki](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#optional-parameter-in-a-param-annotation), and they all seem to suggest that your style is the right one.

The rest of our code base uses a nullable type instead of `undefined`
(e.g., `{?MDCTextFieldBottomLineFoundation}`), but that's technically incorrect, and our internal Closure Compiler usually complains about it.

We should probably update the rest of our components to use your style of annotations at some point.